TypeError: in method 'doppler_correction_make'

Hello!

When attempting to run the example flowgraph **doppler_correction_example.grc**, I get the following:

```
Traceback (most recent call last):
  File "/home/k4kdr/grc/doppler_correction_example.py", line 228, in <module>
    main()
  File "/home/k4kdr/grc/doppler_correction_example.py", line 204, in main
    tb = top_block_cls()
  File "/home/k4kdr/grc/doppler_correction_example.py", line 81, in __init__
    self.satellites_doppler_correction_0 = satellites.doppler_correction('/home/k4kdr/gr-satellites/examples/doppler_correction/doppler_file.txt', samp_rate, 0)
  File "/usr/local/lib/python3/dist-packages/satellites/satellites_swig.py", line 4217, in make
    return _satellites_swig.doppler_correction_make(filename, samp_rate, t0)
TypeError: in method 'doppler_correction_make', argument 1 of type 'std::string &'

>>> Done (return code 1)
```

... I have searched online but so far have not found anything to indicate what is causing this.  

My relevant system info:

python --version
**Python 3.6.9**

swig -version
**SWIG Version 4.0.2**

uname -a
**Linux 3010i5 5.4.0-124-generic #140~18.04.1-Ubuntu SMP Fri Aug 5 11:43:34 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux**

Thank you!

Thanks so much for checking on this!!

The "make" step fails as shown below:

```
[ 94%] Building CXX object lib/CMakeFiles/gnuradio-satellites.dir/viterbi/viterbi.cc.o
[ 95%] Linking CXX shared library libgnuradio-satellites.so
[ 95%] Built target gnuradio-satellites
[ 95%] Built target satellites_swig_swig_compilation
[ 96%] Building CXX object swig/CMakeFiles/satellites_swig.dir/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx.o
/home/k4kdr/gr-sats-swig-test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1847:36: warning: cast between incompatible function types from ‘PyObject* (*)(PyObject*)’ {aka ‘_object* (*)(_object*)’} to ‘PyCFunction’ {aka ‘_object* (*)(_object*, _object*)’} [-Wcast-function-type]
   {(char *)"disown",  (PyCFunction)SwigPyObject_disown,  METH_NOARGS,  (char *)"releases ownership of the pointer"},
                                    ^~~~~~~~~~~~~~~~~~~
/home/k4kdr/gr-sats-swig-test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1848:36: warning: cast between incompatible function types from ‘PyObject* (*)(PyObject*)’ {aka ‘_object* (*)(_object*)’} to ‘PyCFunction’ {aka ‘_object* (*)(_object*, _object*)’} [-Wcast-function-type]
   {(char *)"acquire", (PyCFunction)SwigPyObject_acquire, METH_NOARGS,  (char *)"acquires ownership of the pointer"},
                                    ^~~~~~~~~~~~~~~~~~~~
/home/k4kdr/gr-sats-swig-test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1851:36: warning: cast between incompatible function types from ‘PyObject* (*)(PyObject*)’ {aka ‘_object* (*)(_object*)’} to ‘PyCFunction’ {aka ‘_object* (*)(_object*, _object*)’} [-Wcast-function-type]
   {(char *)"next",    (PyCFunction)SwigPyObject_next,    METH_NOARGS,  (char *)"returns the next 'this' object"},
                                    ^~~~~~~~~~~~~~~~~
/home/k4kdr/gr-sats-swig-test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1852:36: warning: cast between incompatible function types from ‘PyObject* (*)(SwigPyObject*)’ {aka ‘_object* (*)(SwigPyObject*)’} to ‘PyCFunction’ {aka ‘_object* (*)(_object*, _object*)’} [-Wcast-function-type]
   {(char *)"__repr__",(PyCFunction)SwigPyObject_repr,    METH_NOARGS,  (char *)"returns object representation"},
                                    ^~~~~~~~~~~~~~~~~
/home/k4kdr/gr-sats-swig-test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx: In function ‘PyObject* _wrap_doppler_correction_make(PyObject*, PyObject*, PyObject*)’:
/home/k4kdr/gr-sats-swig-test/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:36920:72: error: no matching function for call to ‘gr::satellites::doppler_correction::make(std::__cxx11::string&, double&, double&)’
       result = gr::satellites::doppler_correction::make(*arg1,arg2,arg3);
```

That's weird. I've just tested again and this build for me. I get some warnings, but not the "no matching function for call" error:

```
[ 95%] Linking CXX shared library libgnuradio-satellites.so
[ 95%] Built target gnuradio-satellites
Scanning dependencies of target satellites_swig
[ 96%] Building CXX object swig/CMakeFiles/satellites_swig.dir/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx.o
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx: In function 'PyTypeObject* SwigPyObject_TypeOnce()':
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1770:5: warning: missing initializer for member '_typeobject::tp_vectorcall' [-Wmissing-field-initializers]
 1770 |     };
      |     ^
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1770:5: warning: missing initializer for member '_typeobject::tp_print' [-Wmissing-field-initializers]
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx: In function 'PyTypeObject* SwigPyPacked_TypeOnce()':
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1931:5: warning: missing initializer for member '_typeobject::tp_vectorcall' [-Wmissing-field-initializers]
 1931 |     };
      |     ^
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:1931:5: warning: missing initializer for member '_typeobject::tp_print' [-Wmissing-field-initializers]
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx: In function 'PyTypeObject* swig_varlink_type()':
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:97303:7: warning: missing initializer for member '_typeobject::tp_vectorcall' [-Wmissing-field-initializers]
97303 |       };
      |       ^
/gr-satellites/build/swig/CMakeFiles/satellites_swig.dir/satellites_swigPYTHON_wrap.cxx:97303:7: warning: missing initializer for member '_typeobject::tp_print' [-Wmissing-field-initializers]
[ 97%] Linking CXX shared module _satellites_swig.so
[ 97%] Built target satellites_swig
Scanning dependencies of target pygen_swig_b71b8
[100%] Generating satellites_swig.pyc
[100%] Generating satellites_swig.pyo
[100%] Built target pygen_swig_b71b8
```

I think that somehow you have a leftover from the old code, since your error is mentioning `std::__cxx11::string&`, which is the thing I replaced by `const char*`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 - Python bindings for 8APSK Costas Loop
+- Python bindings for Doppler correction block under GNU Radio 3.8
 
 ## [5.0.0], [4.7.0], [3.14.0] - 2022-07-22
 

--- a/include/satellites/doppler_correction.h
+++ b/include/satellites/doppler_correction.h
@@ -54,7 +54,7 @@ public:
      * \param samp_rate Sample rate
      * \param t0 Timestamp corresponding to the first sample
      */
-    static sptr make(std::string& filename, double samp_rate, double t0);
+    static sptr make(const char* filename, double samp_rate, double t0);
 };
 
 } // namespace satellites

--- a/lib/doppler_correction_impl.cc
+++ b/lib/doppler_correction_impl.cc
@@ -18,14 +18,14 @@ namespace gr {
 namespace satellites {
 
 doppler_correction::sptr
-doppler_correction::make(std::string& filename, double samp_rate, double t0)
+doppler_correction::make(const char* filename, double samp_rate, double t0)
 {
     return gnuradio::get_initial_sptr(
         new doppler_correction_impl(filename, samp_rate, t0));
 }
 
 
-doppler_correction_impl::doppler_correction_impl(std::string& filename,
+doppler_correction_impl::doppler_correction_impl(const char* filename,
                                                  double samp_rate,
                                                  double t0)
     : gr::sync_block("doppler_correction",
@@ -43,7 +43,7 @@ doppler_correction_impl::doppler_correction_impl(std::string& filename,
 
 doppler_correction_impl::~doppler_correction_impl() {}
 
-void doppler_correction_impl::read_doppler_file(std::string& filename)
+void doppler_correction_impl::read_doppler_file(const char* filename)
 {
     std::ifstream input_file(filename);
     double time;

--- a/lib/doppler_correction_impl.h
+++ b/lib/doppler_correction_impl.h
@@ -39,10 +39,10 @@ private:
             d_phase += 2 * GR_M_PI;
     }
 
-    void read_doppler_file(std::string& filename);
+    void read_doppler_file(const char* filename);
 
 public:
-    doppler_correction_impl(std::string& filename, double samp_rate, double t0);
+    doppler_correction_impl(const char* filename, double samp_rate, double t0);
     ~doppler_correction_impl();
 
     int work(int noutput_items,

